app/models/manageiq/providers/ansible_tower/automation_manager/credential/api_create.rb

```diff
@@ -0,0 +1,36 @@
+class ManageIQ::Providers::AnsibleTower::AutomationManager::Credential
+  module ApiCreate
+    def create_in_provider(manager_id, params)
+      manager = ExtManagementSystem.find(manager_id)
+      credential = manager.with_provider_connection do |connection|
+        connection.api.credentials.create!(params)
+      end
+
+      # Get the record in our database
+      # TODO: This needs to be targeted refresh so it doesn't take too long
+      EmsRefresh.queue_refresh(manager, nil, true) if !manager.missing_credentials? && manager.authentication_status_ok?
```

Comment:
Typically CRUD of native is done inside of a state machine like how provisioning does it where there is a step to create natively, poke the refresh asynchronously, then transition to a state where you can check asynchronously.  All of these seems to be happening directly in this method, which means a worker is going to be completely locked waiting for refresh to complete.

I feel like a better approach for *ANY* CRUD of native objects is to use some sort of MiqTask, but not as it's done here.  This task should call a method that natively creates the object, then pokes the refresh and then immediately returns.  The MiqTask is then responsible for the next step of seeing if the refresh has completed.  Once the MiqTask completes, it can use UI Notifications to tell the user that the object is ready.

So, for the use case of creating a credential and then using it during creating a repository, I would see:

- User creates the credential
- Creation request put in queue via a task and user is notified that "Request to create credential has started"
- Queue picks up the message, executes the native create, pokes the EmsRefresh
- Task then puts polling on the queue until the expected object is in the database.
- Task then notifies via UI notification, and marks itself as done.
- The User will then see the notification and can go create his repository.  If he tries any earlier, the new cred just won't appear.